### PR TITLE
feature: facebook login

### PR DESCRIPTION
**Pull request by SergeyMarynenko**

### CODE REVIEW

File: config/app.php [hunk 1]
```diff
@@ -166,6 +166,7 @@
 
 
 		Modules\Core\Providers\AsgardServiceProvider::class,
+        Laravel\Socialite\SocialiteServiceProvider::class,
```
Comment (7s4r) [inline config/app.php]: Bad indentation

File: config/services.php [hunk 2]
```diff
@@ -34,4 +34,11 @@
 		'secret' => '',
 	],
 
+
+    'facebook' => [
```
Comment (7s4r) [inline config/services.php]: bad indentation

File: scrutinizer.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@ build:
         php: '7.1'
     tests:
         override:
-            - composer require --dev phpunit/phpunit
+            - composer require --dev phpunit/phpunit:~4.0
```
Comment (7s4r) [inline scrutinizer.yml]: add space after `:`

File: Modules/User/Http/Controllers/AuthController.php [hunk 4]
```diff
@@ -26,6 +29,57 @@ public function __construct()
...
+
+        return redirect('/');
+    }
+
+
+    private function findOrCreateUser($facebookUser){
+        $authUser = Ruser::where('facebook_id', $facebookUser->id)->first();
+        $authEmail = Ruser::where('email', $facebookUser->email)->first();
+
+        if ($authUser){
```
Comment (7s4r) [inline Modules/User/Http/Controllers/AuthController.php]: add space before `{`

File: Modules/User/Http/Controllers/AuthController.php [hunk 5]
```diff
@@ -26,6 +29,57 @@ public function __construct()
...
+            'email' => $authUser->email,
+        ];
+
+        $this->auth->login($credentials, true);
+
+        return redirect('/');
+    }
+
+
+    private function findOrCreateUser($facebookUser){
```
Comment (7s4r) [inline Modules/User/Http/Controllers/AuthController.php]: add space before `{`

File: Modules/User/Http/Controllers/AuthController.php [hunk 6]
```diff
@@ -26,6 +29,57 @@ public function __construct()
 
     }
 
+    public function redirect(){
+        return Socialite::driver('facebook')->redirect();
+    }
+
+
+    public function callback(){
```
Comment (7s4r) [inline Modules/User/Http/Controllers/AuthController.php]: add space before `{`

File: Modules/User/Http/Controllers/AuthController.php [hunk 7]
```diff
@@ -26,6 +29,57 @@ public function __construct()
 
     }
 
+    public function redirect(){
```
Comment (7s4r) [inline Modules/User/Http/Controllers/AuthController.php]: add space before `{`

File: Modules/User/Http/Controllers/AuthController.php [hunk 8]
```diff
@@ -26,6 +29,57 @@ public function __construct()
...
+    private function findOrCreateUser($facebookUser){
+        $authUser = Ruser::where('facebook_id', $facebookUser->id)->first();
+        $authEmail = Ruser::where('email', $facebookUser->email)->first();
+
+        if ($authUser){
+            return $authUser;
+        }
+
+
+        if($authEmail){
```
Comment (7s4r) [inline Modules/User/Http/Controllers/AuthController.php]: add space before `{` and after `if`

File: Modules/User/Http/Controllers/AuthController.php [hunk 9]
```diff
@@ -26,6 +29,57 @@ public function __construct()
...
+            $authEmail->facebook_id = $facebookUser->id;
+            $authEmail->save();
+            return $authEmail;
+        } else {
+
```
Comment (7s4r) [inline Modules/User/Http/Controllers/AuthController.php]: Useless space

File: Modules/User/Http/Controllers/AuthController.php [hunk 10]
```diff
@@ -26,6 +29,57 @@ public function __construct()
...
+            return User::create([
+                'email' => $facebookUser->email,
+                'facebook_id' => $facebookUser->id,
+                'first_name' => $superaname[0],
+                'last_name' => $superaname[1],
+            ]);
+        }
+    }
+
+
```
Comment (7s4r) [inline Modules/User/Http/Controllers/AuthController.php]: Useless space

File: Modules/User/Http/Controllers/AuthController.php [hunk 11]
```diff
@@ -26,6 +29,57 @@ public function __construct()
 
     }
 
+    public function redirect(){
+        return Socialite::driver('facebook')->redirect();
+    }
+
+
```
Comment (7s4r) [inline Modules/User/Http/Controllers/AuthController.php]: Useless space

File: Modules/User/Http/Controllers/AuthController.php [hunk 12]
```diff
@@ -1,6 +1,9 @@
 <?php namespace Modules\User\Http\Controllers;
 
+use Exception;
 use Illuminate\Foundation\Bus\DispatchesJobs;
+use Illuminate\Support\Facades\Auth;
+use Laravel\Socialite\Facades\Socialite;
```
Comment (7s4r) [inline Modules/User/Http/Controllers/AuthController.php]: Add dependencies to composer

File: scrutinizer.yml (hunk 3 above)
Comment (SergeyMarynenko) [inline scrutinizer.yml]: no its wrong if i add space

File: config/services.php [hunk 13]
```diff
@@ -34,4 +34,10 @@
 		'secret' => '',
 	],
 
+    'facebook' => [
+        'client_id' => '631817180357642',
+        'client_secret' => '84da949c6a6f095a5804e4e4cfe8c34b',
+        'redirect' => 'http://localhost:8000/en/user/callback',
```
Comment (7s4r) [inline config/services.php]: dynamize this url

**Pull request merged**